libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp

```diff
@@ -45,11 +45,12 @@ void NavEKF3_core::ResetVelocity(void)
             // clear the timeout flags and counters
             velTimeout = false;
             lastVelPassTime_ms = imuSampleTime_ms;
-        } else if ((imuSampleTime_ms - extNavVelMeasTime_ms < 250 && posResetSource == DEFAULT) || velResetSource == EXTNAV) {
+        } else if ((imuSampleTime_ms - extNavVelMeasTime_ms < 250 && velResetSource == DEFAULT) || velResetSource == EXTNAV) {
             // use external nav data as the 2nd preference
             // already corrected for sensor position
-            stateStruct.velocity = extNavVelDelayed.vel;
-            P[5][5] = P[4][4] = P[3][3] = sq(extNavVelDelayed.err);
+            stateStruct.velocity.x = extNavVelDelayed.vel.x;
+            stateStruct.velocity.y = extNavVelDelayed.vel.y;
+            P[5][5] = P[4][4] = sq(extNavVelDelayed.err);
```

Comment:
These P[][] numbers are all a bit confusing.  I clearly copy-pasted the original line from [here in the EKF2](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp#L52) but it seems like perhaps P[2][2] should hold the expeted error for velocity Z?

Reply:
See https://github.com/ArduPilot/ardupilot/blob/2eeac9e6fd44805962daf322b068eeedf794db20/libraries/AP_NavEKF3/AP_NavEKF3_core.h#L508-L525 for information on indexing for state vector

NED velocity is in indexes 3,5,6

Reply:
Indexes in P matrix must be consistent with state vector.

Reply:
OK, so I think this means we should be setting P[6][6] with the velocity Z-axis error..

Reply:
It is set in `ResetHeight()` like GPS currently did.

Reply:
Maybe we could change z velocity reset with ext nav vel to `ResetVelocity()`?